django-verdant/rca/migrations/0087_staffpage_migrate_area_data_to_roles.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2017-07-06 21:26
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import Count
+
+
+class Migration(migrations.Migration):
+    def move_area_data(apps, schema_editor):
+        """
+        Move StaffPage.area to StaffPageRole.area if the user is
+        academic/administrative and if StaffPageRole.area is empty.
+        """
+        StaffPage = apps.get_model('rca', 'StaffPage')
+
+        # Move area data to roles
+        records = StaffPage.objects \
+                            .prefetch_related('roles') \
+                            .annotate(roles_count=Count('roles__id')) \
+                            .filter(staff_type__in=('administrative', 'academic')) \
+                            .filter(area__isnull=False) \
+                            .filter(roles_count__gt=0)
+
+        for staff in records:
+            first_role = staff.roles.filter(area__isnull=True).first()
+            if first_role is not None:
+                first_role.area = staff.area
```

Comment:
Don't we need to copy the location field too from the StaffPage? Maybe only if it has one role only?